cmd/agentbasedinstaller/client/main.go

```diff
@@ -35,13 +35,15 @@ import (
 
 	"github.com/kelseyhightower/envconfig"
 	"github.com/openshift/assisted-service/client"
+	"github.com/openshift/assisted-service/pkg/auth"
 	log "github.com/sirupsen/logrus"
 )
 
 const failureOutputPath = "/var/run/agent-installer/host-config-failures"
 
 var Options struct {
 	ServiceBaseUrl string `envconfig:"SERVICE_BASE_URL" default:""`
+	Token          string `envconfig:"PULL_SECRET_TOKEN" default:""`
```

Comment:
The environment variable name here in the ABI client doesn't need to match the name in agent.service even if it's the same value.

The agent env var is named `PULL_SECRET_TOKEN` because for the SaaS the token is indeed based on the pull secret. For on-prem we just reused the variable in the agent because it did what we needed it to do without changes.

We should try to name this as well as possible even if it is eventually used in other places under different names.